data/sqldb.go

```diff
@@ -0,0 +1,315 @@
+package data
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/farshidtz/senml/v2"
+	"github.com/linksmart/historical-datastore/common"
+	"github.com/linksmart/historical-datastore/registry"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// SqlStorage implements a SqlDB storage client for HDS Data API
+type SqlStorage struct {
+	pool *sql.DB
+}
+
+func NewSqlStorage(conf common.DataConf) (storage *SqlStorage, disconnect_func func() error, err error) {
+	storage = new(SqlStorage)
+	storage.pool, err = sql.Open("sqlite3", conf.Backend.DSN) //In future, this shall be made configurable.
+	if err != nil {
+		return nil, nil, err
+	}
+	return storage, storage.Disconnect, err
+}
+
+func btoi(b bool) int {
+	if b {
+		return 1
+	}
+	return 0
+}
+
+func (s *SqlStorage) Submit(data map[string]senml.Pack, sources map[string]*registry.DataStream) error {
+	for ds, pack := range data {
+		valueStrings := make([]string, 0, len(pack))
+		valueArgs := make([]interface{}, 0, len(pack)*2)
+		switch sources[ds].Type {
+		case registry.Float:
+			for _, r := range pack {
+				valueStrings = append(valueStrings, "(?, ?)")
+				valueArgs = append(valueArgs, r.Time, *r.Value)
+			}
+		case registry.String:
+			for _, r := range pack {
+				valueStrings = append(valueStrings, "(?, ?)")
+				valueArgs = append(valueArgs, r.Time, r.StringValue)
+			}
+		case registry.Bool:
+			for _, r := range pack {
+				valueStrings = append(valueStrings, "(?, ?)")
+				valueArgs = append(valueArgs, r.Time, btoi(*r.BoolValue))
+			}
+		case registry.Data:
+			for _, r := range pack {
+				valueStrings = append(valueStrings, "(?, ?)")
+				valueArgs = append(valueArgs, r.Time, r.DataValue)
+			}
+		}
+
+		stmt := fmt.Sprintf("REPLACE INTO [%s] (time, value) VALUES %s",
+			ds, strings.Join(valueStrings, ","))
+		_, err := s.pool.Exec(stmt, valueArgs...)
+		return err
+	}
+
+	return nil
+}
+
+//This function converts a floating point number (which is supported by senml) to a bytearray
```

Comment:
is this returning "bytearray"?  Also check the comment on the next function.

Reply:
was wrong. correcting it.